### PR TITLE
add rest API to query aggregated metrics

### PR DESCRIPTION
**Pull request by boyang9527**
[#158417452]

### CODE REVIEW

File: src/autoscaler/eventgenerator/server/eventgen_handler.go [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+}
+
+func (h *EventGenHandler) GetAggregatedMetricHistories(w http.ResponseWriter, r *http.Request, vars map[string]string) {
+	appID := vars["appid"]
+	metricType := vars["metrictype"]
+	startParam := r.URL.Query()["start"]
+	endParam := r.URL.Query()["end"]
+	orderParam := r.URL.Query()["order"]
+
+	h.logger.Debug("get-aggregateed-metric-histories", lager.Data{"appid": appID, "metrictype": metricType, "start": startParam, "end": endParam, "order": orderParam})
```
Comment (qibobo) [inline src/autoscaler/eventgenerator/server/eventgen_handler.go]: aggregateed -> aggregated?

File: src/autoscaler/eventgenerator/server/eventgen_handler.go [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+	var err error
+	start := int64(0)
+	end := int64(-1)
+	order := db.ASC
+
+	if len(startParam) == 1 {
+		start, err = strconv.ParseInt(startParam[0], 10, 64)
+		if err != nil {
+			h.logger.Error("get-aggregateed-metric-histories-parse-start-time", err, lager.Data{"start": startParam})
```
Comment (qibobo) [inline src/autoscaler/eventgenerator/server/eventgen_handler.go]: the same as above

File: src/autoscaler/eventgenerator/server/eventgen_handler.go [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+			handlers.WriteJSONResponse(w, http.StatusBadRequest, models.ErrorResponse{
+				Code:    "Bad-Request",
+				Message: "Error parsing start time"})
+			return
+		}
+	} else if len(startParam) > 1 {
+		h.logger.Error("get-aggregateed-metric-histories-get-start-time", err, lager.Data{"start": startParam})
```
Comment (qibobo) [inline src/autoscaler/eventgenerator/server/eventgen_handler.go]: the same as above

File: src/autoscaler/eventgenerator/server/eventgen_handler.go [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+		handlers.WriteJSONResponse(w, http.StatusBadRequest, models.ErrorResponse{
+			Code:    "Bad-Request",
+			Message: "Incorrect start parameter in query string"})
+		return
+	}
+
+	if len(endParam) == 1 {
+		end, err = strconv.ParseInt(endParam[0], 10, 64)
+		if err != nil {
+			h.logger.Error("get-aggregateed-metric-histories-parse-end-time", err, lager.Data{"end": endParam})
```
Comment (qibobo) [inline src/autoscaler/eventgenerator/server/eventgen_handler.go]: the same as above and there are some below.

File: src/autoscaler/eventgenerator/server/server_test.go [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+		})
+
+		It("should return 404", func() {
+			Expect(err).ToNot(HaveOccurred())
+			Expect(rsp.StatusCode).To(Equal(http.StatusNotFound))
+			rsp.Body.Close()
+		})
+	})
+
+	Context("when using wrong method to retrieve aggreated metric histories", func() {
```
Comment (qibobo) [inline src/autoscaler/eventgenerator/server/server_test.go]: what is the difference between this one and the one above?

File: src/autoscaler/eventgenerator/config/config_test.go [hunk 6]
```diff
@@ -1073,78 +1091,6 @@ defaultBreachDurationSecs: NOT-INTEGER-VALUE
...
-					DefaultStatWindowSecs:     300,
-					CircuitBreaker: CircuitBreakerConfig{
-						BackOffInitialInterval:  DefaultBackOffInitialInterval,
-						BackOffMaxInterval:      DefaultBackOffMaxInterval,
-						ConsecutiveFailureCount: DefaultBreakerConsecutiveFailureCount,
-					},
-				}))
-			})
-		})
-
```
Comment (qibobo) [inline src/autoscaler/eventgenerator/config/config_test.go]: why remove the test?

File: src/autoscaler/eventgenerator/server/server_test.go (hunk 5 above)
Comment (boyang9527) [reply src/autoscaler/eventgenerator/server/server_test.go]: it is a duplicate, will remove

File: src/autoscaler/eventgenerator/config/config_test.go (hunk 6 above)
Comment (boyang9527) [reply src/autoscaler/eventgenerator/config/config_test.go]: it is a duplicate of test case:

https://github.com/boyang9527/app-autoscaler/blob/d0c94cab91e8838ba887dde90830a8eb3e0d5ccf/src/autoscaler/eventgenerator/config/config_test.go#L218

**Comment by qibobo** [comment]
LGTM

**Comment by cdlliuy** [comment]
LGTM

**Pull request merged**